index.js

```diff
@@ -32,6 +32,38 @@ function getMainArgs() {
   return process.argv.slice(2);
 }
 
+function setOptionValue(parseOptions, option, value, result) {
+  const multiple = parseOptions.multiples &&
+    parseOptions.multiples.includes(option);
+  const withValue = parseOptions.withValue &&
+    parseOptions.withValue.includes(option);
+
+  // Normal flag: !withValue && value === undefined
+  // Normal value, withValue && value !== undefined
+  // Special case: withValue && value === undefined
+  //    store as normal for withValue with value undefined
+  // Special case: !withValue && value !== undefined
+  //    store as normal for withValue (and not a flag)
+
+  // Flags
+  // Only mark flags for plain flag without a value, expected or otherwise.
+  const isFlag = !withValue && value === undefined;
+  if (isFlag)
+    result.flags[option] = true;
+
+  // Values
+  if (multiple) {
+    // Always store value in array, including for flags.
+    const val = isFlag ? true : value;
+    if (result.values[option] !== undefined)
+      result.values[option].push(val);
```

Comment:
This line assumes that it’s an array, but the previous line isn’t checking that it’s an array.

Reply:
The context is that the option is a multiple, and the only values come through this block of code. If the option already has a value it comes from the next line, which sticks the first value into an array. Hence assuming it is an array and ok to `push`.

However, I thought about it when writing the code, and you asked too. Assuming you agree with my logic, I should add a comment to clarify.

Reply:
Added comment.